x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade.md

```diff
@@ -500,242 +238,219 @@ And user should see a CTA to upgrade prebuilt rules
+#### **Scenario: User can upgrade only conflict-free rules when attempting to upgrade all rules**
 
-**Automation**: unit tests.
+**Automation**: 1 e2e test with mock rules
 
 ```Gherkin
-Given prebuilt rules package is not installed
-When user opens the Add Rules page
-Then user should see a loading skeleton until the package installation is completed
+Given X prebuilt rules are installed in Kibana
+And for Y of the installed rules there are new versions available
+And Z (where Z < Y) rules have conflicts with the current versions
+And user is on the Rule Updates table
+Then Y rules available for upgrade should be displayed in the table
+Then user should see an enabled CTA to upgrade all rules
+When user clicks the CTA
+A modal window should inform the user that only K (where K < Y) rules without conflicts will be upgraded
+When user confirms the action in the modal
+Then success message should be displayed after upgrade informing that K rules were updated
+And the K upgraded rules should be removed from the table
+And the remaining M = Y - K rules should still be present in the table
+And user should see the number of rules available to upgrade decreased by K number of upgraded rules
 ```
 
-#### **Scenario: User can navigate from the Add Rules page to the Rule Management page via breadcrumbs**
 
-**Automation**: 1 e2e test.
+### Rule upgrade workflow: upgrading rules with rule type change
 
-```Gherkin
-Given user is on the Add Rules page
-When user navigates to the Rule Management page via breadcrumbs
-Then the Rule Management page should be displayed
-```
-
-### Rule upgrade workflow: base cases
-
-#### **Scenario: User can upgrade prebuilt rules one by one**
+#### **Scenario: User can upgrade rule with rule type change individually**
 
-**Automation**: 1 e2e test with mock rules + integration tests with mock rules that would test /status and /upgrade/\* endpoints in integration.
+**Automation**: 1 e2e test with mock rules
 
 ```Gherkin
-Given X prebuilt rules are installed in Kibana
-And for Y of the installed rules there are new versions available
-And user is on the Rule Management page
+Given a prebuilt rule is installed in Kibana
+And this rule has an update available that changes its rule type
 When user opens the Rule Updates table
-Then Y rules available for upgrade should be displayed in the table
-When user upgrades one individual rule without previewing it
+Then this rule should be displayed in the table
+And the Upgrade Rule button should be enabled
+When user upgrades the rule
 Then success message should be displayed after upgrade
 And the upgraded rule should be removed from the table
 And user should see the number of rules available to upgrade decreased by 1
 ```
 
-#### **Scenario: User can upgrade multiple prebuilt rules selected on the page**
+#### **Scenario: User can bulk upgrade selected rules with rule type changes**
```

Comment:
Users can upgrade rules without customizations with rule type changes with disabled feature flag but it's not possible with enabled feature flag. The only way to upgrade rules with type changes is via rule upgrade flyout.